### ISSUE TITLE
Matrix test unexpectedly failing

### ISSUE THREAD

**Issue by MTWhyte** [libsemigroups/libsemigroups#395]
On the current `main`, when I run `./test_matrix`, I get the following error output:

```
Matrix 011: MaxPlusMat<3>
-------------------------------------------------------------------------------
tests/test-matrix.cpp:946
...............................................................................

tests/test-matrix.cpp:644: FAILED:
  REQUIRE( y == x )
with expansion:
  {{-2147483648, -2147483648, -2147483648}, {-2147483648, -2147483648, -
  2147483648}, {-2147483648, -2147483648, -2147483648}}
  ==
  {{-2, 2, 0}, {-1, 0, 0}, {1, -3, 1}}

Matrix 011: MaxPlusMat<3> . . .
===============================================================================
test cases:  1 | 1 failed
assertions: 10 | 9 passed | 1 failed
```

**Comment by james-d-mitchell** [comment]
Interestingly, this is now also showing up in the CI at, for example:

https://github.com/libsemigroups/libsemigroups/actions/runs/3336794000/jobs/5522366230

Definitely a bug...

**Comment by james-d-mitchell** [comment]
It seems that this is caused by some change in `clang`, with 

~~~~
Apple clang version 13.1.6 (clang-1316.0.21.2.5)
Target: arm64-apple-darwin21.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
~~~~

the bug occurs (I can reproduce), but it worked fine in an earlier version of clang (12.something).

**Comment by james-d-mitchell** [comment]
Resolved in #401

**Comment by james-d-mitchell** [comment]
Resolved in v2.3.2
